pkg/controllers/reconciler.go

```diff
@@ -39,6 +39,7 @@ import (
 var (
 	UpgradePlugins   bool
 	UpgradeToVersion string
+	EnvtestMode      bool
```

Comment:
> Apparently controller-runtime caches read operations,
> injecting a daemonset for upgrade envtest is problematic without it.

how this works in your manual testing?

Reply:
@mythi 
> how this works in your manual testing?

Once the controller is instantiated, it does some indexing to cache read operations. In manual tests it is able to find the previous daemonset.

With the injection of the daemonset under envtest, apparently not. I am not entirely sure what's the subtle difference, either the controller should be deleted and recreated before each upgrade envtest (didn't succeed with this) or some subtle detail in indexing.

Reply:
@ozhuraki would this type of test work better under the e2e test framework then? I don't like the fact that we need to modify the controller's behavior depending on whether it's under test or not.